xarray/tests/test_variable.py

```diff
@@ -344,9 +344,10 @@ def test_pandas_period_index(self):
         assert v[0] == pd.Period("2000", freq="B")
         assert "Period('2000-01-03', 'B')" in repr(v)
 
-    def test_1d_math(self):
-        x = 1.0 * np.arange(5)
-        y = np.ones(5)
+    @pytest.mark.parametrize("dtype", [float, int])
+    def test_1d_math(self, dtype):
```

Comment:
```suggestion
    def test_1d_math(self, dtype: np.typing.DTypeLike) -> None:
```

Reply:
Adding type checking on `test_1d_math` triggered attribute errors on existing lines of the test.  
`xarray/tests/test_variable.py:370: error: "ndarray[Any, dtype[bool_]]" has no attribute "values" `

I'm not sure of the simple fix so removed it from that test.

Reply:
Not sure why numpy takes priority when it's first in the operation. 
If you don't want to work on it, add the typing back but add ignores to the bad lines:

```python
xv.values  # type: ignore[attr-defined] # TODO: Fix mypy thinking numpy takes priority, GH7780
```